### ISSUE TITLE
Concern about Travis CI test

### ISSUE THREAD

**Issue by LRossman** [OpenWaterAnalytics/EPANET#405]
On my latest PR the Travis CI test said that the PR passed yet its output log showed the following:
```
*** Error in `/home/travis/build/OpenWaterAnalytics/EPANET/buildprod/bin/runepanet': double free or corruption (!prev): 0x00000000007eca20 ***
example1: fail
example1: Non-zero exit code
*** Error in `/home/travis/build/OpenWaterAnalytics/EPANET/buildprod/bin/runepanet': corrupted size vs. prev_size: 0x0000000001aa0000 ***
example3: fail
example3: Non-zero exit code
```
There were similar failures for three other test files. Each of these files ran with no problems on my local machine. Does this indicate an issue with the Travis CI testing procedure?

**Comment by michaeltryby** [comment]
@LRossman Travis CI says your last commit to your PR failed. Perhaps you aren't looking at the results for the most current build. By the looks of it, the changes you are making are crashing hard. 

Why don't you try building with mingw locally? That's what I do when I need to root out problems on linux/gcc.

**Comment by LRossman** [comment]
@michaeltryby my comment was referring to the commit prior to the last one that failed (Travis build #620, not #622). The latest commit, that closes the output file when `EN_closeQ` is called and opens it for `EN_initQ` and `EN_report`, has the re-entrancy test pass on AppVeyor but fail on Travis. Any ideas why? I will try building locally with MinGW.

I believe it's really important that client apps have direct access to the output results file without having to close down the entire project. In lieu of this we could add two new functions, `EN_getnoderesult` and `EN_getlinkresult`, that return a computed result for a given variable at a given time period. However this option probably wouldn't be as efficient as having direct access to the file.

If a client app can't get access to all of the results of a run (while the project is kept open) it would have to capture results at each reporting time period (using the existing  `EN_getnodevalue` and `EN_getlinkvalue`) as the simulation unfolds and save them somewhere so that interactive display of the results (in graphs and tables) could be done efficiently.

**Comment by michaeltryby** [comment]
It’s important to remember that Travis isn’t doing anything magical. It’s just building and running EPANET on Ubuntu Linux. Likewise, reg testing is just running the EPANET CLI for a bunch of known test cases.

**Comment by michaeltryby** [comment]
@LRossman What happened when you built using mingw locally? Were you able to reproduce the problem?

**Comment by LRossman** [comment]
@michaeltryby building with mingw didn't reproduce the problem which was a runtime error not a build error. I assume that Travis runs on a Linux system while my mingw build runs on Windows. Looking at the error report ("double free or corruption") and since my code changes involved using `fclose` on the binary output file, I think I may have been freeing a file handle twice.

**Comment by LRossman** [comment]
The commit in question was trying to close an already closed file (due to two file handles pointing to the same file). This produced a runtime error under Linux (hence the Travis CI failure), but not under Windows (allowing AppVeyor to succeed). This issue has been fixed in PR #408.
